### DISCUSSION TITLE
error with no shm

### DISCUSSION THREAD

**Discussion by railgauge** [audacity/audacity#4324] [Q&A]
Using a chroot linux environment on an Android device and starting Audacity causes the following GUI error message to appear:
<pre>
------------------------------------------------
|         Audacity Startup Failure             |
|    Unable to create shared memory segment    |
|  error code=38 : "Function not implemented". |
------------------------------------------------
</pre>
I can see this error comes from AudacityApp.cpp. Is /dev/shm strictly required for Audacity or could there be a workaround? I understand that shm is not supported on Android.

Device information:
```
OS (host):            Android 13
OS (chroot):          Arch Linux aarch64
Kernel:               5.10.107
Architecture:         aarch64
CPU op-mode(s):       32-bit, 64-bit
Byte Order:           Little Endian
```

**Comment by int-ua** [comment]
This use-case is about to become mainstream with Android Virtualization Framework and Linux Terminal on Google Pixel devices.

**Reply by kryksyh** [reply]
@int-ua Audacity 4 uses a different approach for ensuring single-instance operations, so it should not be an issue

**Reply by int-ua** [reply]
Great. For now it's 3.7.3 in the Debian Trixie repos that are used by default.
